Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -2326,6 +2326,11 @@ bool LocalFrameView::scrollToFragment(const URL& url)
             
             if (highlightRanges.size()) {
                 auto range = highlightRanges.first();
+                auto* commonAncestor = commonInclusiveAncestor<ComposedTree>(range);
+                while (commonAncestor && !is<Element>(commonAncestor))
+                    commonAncestor = commonAncestor->parentElement();
```

Comment:
Surely, we need to call this function at most once?

Reply:
Makes sense, done.